app/models/miq_group.rb

```diff
@@ -282,6 +282,18 @@ def self.display_name(number = 1)
     n_('Group', 'Groups', number)
   end
 
+  def delete_from_dashboard_order(widget_set_id)
+    return if settings.blank?
+
+    settings[:dashboard_order]&.delete(widget_set_id)
+  end
+
+  def add_to_dashboard_order(widget_set_id)
+    self.settings ||= {:dashboard_order => []}
+    self.settings[:dashboard_order] ||= []
+    self.settings[:dashboard_order] |= [widget_set_id]
```

Comment:
yes it is logic from UI,
suggested change is not needed:
```ruby
irb(main):004:0> a=[1];a+=[1];b=[2];b|=[2];puts "a:#{a.count} b:#{b.count}";
a:2 b:1
```

Reply:
@kbrock 

> from the method names I had assumed it did nothing if called multiple times. But instead it looks like it will change the data every time.

> Do we really want to update the set data automatically on every save?

We are not updating data every time - thanks to this guard [condition](https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_widget.rb#L330).